numba/np/npyfuncs.py

```diff
@@ -1473,6 +1473,13 @@ def np_datetime_isfinite_impl(context, builder, sig, args):
     return builder.icmp_unsigned('!=', args[0], npdatetime.NAT)
 
 
+def np_isnat_impl(context, builder, sig, args):
+    _check_arity_and_homogeneity(sig, args, 1, return_type=types.boolean)
+    return builder.or_(
+        builder.icmp_unsigned('==', args[0], npdatetime.NAT),
+        builder.icmp_unsigned('==', args[0], npdatetime.NAT_BIG_ENDIAN))
+
```

Comment:
Are you sure this logic is valid? I think a big endian NaT would appear as a valid little endian datetime. e.g.
```python
from numba import njit
import numpy as np

@njit
def foo(a):
    return np.isnat(a)

BENaT = np.datetime64('NaT').byteswap()
arg = np.array(BENaT.view(np.int64), np.dtype('<M8[Y]'))
print("arg", arg)

print(foo.py_func(arg))
print(foo(arg))
```
on a LE machine produces:
```
arg 2098
False
True
```
I'd anticipate that the inverse would hold on a BE machine.

Reply:
Further, it's probably worth checking if any of the other `is*` functions work on BE before going down the route of attempting to support it?

Reply:
What do you mean by _BE_ and _LE_?

Reply:
Hey @stuartarchibald, you're right. My bet is that the [current NumPy implementation](https://github.com/numpy/numpy/blob/0b9b176dcef7f14f56dbd543c89e6da6c6b7b2e4/numpy/core/src/umath/loops.c.src#L1085-L1092) for `np.isnat` does a byte-swapping of the underlying data. Thus, `np.isnat(np.array(['NaT'], dtype='>datetime64[Y]'))` is actually comparing `NPY_DATETIME_NAT == arg.byteswap()`

```python
>>> np.array(['NaT'], dtype='>datetime64[Y]').view(np.int64)
array([128])

>>> np.array(['NaT'], dtype='>datetime64[Y]').byteswap().view(np.int64)
array([-9223372036854775808])

>>> np.array(['NaT'], dtype='<datetime64[Y]').view(np.int64)
array([-9223372036854775808])
```